src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,177 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.ArrayList;
+
+public class TaskList {
+    /** The list of tasks. */
+    private ArrayList<Task> userTasks;
+
+    /**
+     * Constructs a new instance of the TaskList with no tasks inside.
+     */
+    public TaskList() {
+        this.userTasks = new ArrayList<>();
```

Comment:
Perhaps including the type of array list you're initialising here. `new ArrayList<Task>()` instead of `new ArrayList<>()`